src/extractor/myexperience/index.ts

```diff
@@ -0,0 +1,426 @@
+      throw new Error(
+        "Tried to extract data from a page without a posting table"
+      );
+
+    if (!options?.startOnCurrent) {
+      const startingPage = await page.$(
+        `${paginationSelector} ul > li:not(.disabled) > a`
+      );
+      if (startingPage) {
+        startingPage.evaluate((x) => x.click());
+        await delay(3000);
+      }
+    }
+    let openedPage: Page | null = null;
+    const popupHandler = async (popup: Page) => {
+      while (openedPage !== null) await delay(100);
+      openedPage = popup;
+    };
+    page.on("popup", popupHandler);
+
+    const done = new Set<string | undefined>();
+    const results: (Posting | PostingError)[] = [];
+
+    try {
+      for (let i = 0; !options?.maxPages || i < options.maxPages; i++) {
+        let plist;
+        for (const table of await page.$$("table")) {
+          const headerRow = await page.$("thead tr");
+          if (!headerRow) continue;
+          const headerTexts = new Map(
+            [
+              ...(
+                await headerRow.$$eval("td, th", (e) => e.map((x) => x.innerText))
+              ).entries(),
+            ].map(([x, y]) => [y, x] as const)
+          );
+          const postingIdIndex = priorityMatch(
+            headerTexts,
+            "job id",
+            "posting id",
+            "jobid",
+            "postingid",
+            "id"
+          );
+          if (postingIdIndex === undefined) continue;
+
+          const results: {
+            id: string | undefined;
+            row: ElementHandle<HTMLTableRowElement>;
+            openClick: ElementHandle<HTMLElement>;
+          }[] = [];
+
+          for (const row of await table.$$("tbody tr")) {
+            const tds = await row.$$("td");
+            let openClick = await tds[0]?.$("a:nth-of-type(2)");
+          
+            // "Apply" as second button for postings not applied to
+            // "Applied" as first (only) button for postings applied to
+            // "view" as first button on the "Applied To" and similar pages
+            if (await getInnerText(openClick) !== "Apply") {
+              openClick = await tds[0]?.$("a");
+              const innerText = await getInnerText(openClick);
+              if (innerText === "view") {
+                // make onclick open in a new tab
+                const onClickValue = await openClick?.evaluate((el) => el.getAttribute('onclick'));
+                if (!onClickValue) continue;
+                const newOnClickValue = onClickValue?.replace("htm').submit()", "htm', '_blank').submit()")
+                await openClick?.evaluate((el, newOnClickValue) => el.setAttribute('onclick', newOnClickValue), newOnClickValue);
+              } else if (innerText !== "Applied") {
+                continue;
+              }
+            }
+            results.push({
+              id: await getInnerText(tds[postingIdIndex]),
+              row,
+              openClick: openClick as ElementHandle<HTMLElement>,
```

Comment:
Arbitrary cast is unnecessary
```suggestion
              openClick: openClick!,
```